google-cloud-bigquery/lib/google/cloud/bigquery/model.rb

```diff
@@ -0,0 +1,722 @@
+          resource? && !resource_partial?
+        end
+
+        ##
+        # @private New Model from a Google API Client object.
+        def self.from_gapi gapi, service
+          new.tap do |f|
+            f.gapi = gapi
+            f.service = service
+          end
+        end
+
+        ##
+        # @private New lazy Model object without making an HTTP request.
+        def self.new_reference project_id, dataset_id, model_id, service
+          raise ArgumentError, "project_id is required" unless project_id
+          raise ArgumentError, "dataset_id is required" unless dataset_id
+          raise ArgumentError, "model_id is required" unless model_id
+          raise ArgumentError, "service is required" unless service
+
+          new.tap do |b|
+            reference_gapi = Google::Apis::BigqueryV2::ModelReference.new(
+              project_id: project_id,
+              dataset_id: dataset_id,
+              model_id:   model_id
+            )
+            b.service = service
+            b.instance_variable_set :@reference, reference_gapi
+          end
+        end
+
+        ##
+        # @private New lazy Model object from a Google API Client object.
+        def self.new_reference_from_gapi gapi, service
+          new.tap do |b|
+            b.service = service
+            b.instance_variable_set :@reference, gapi.model_reference
+          end
+        end
+
+        protected
+
+        ##
+        # Raise an error unless an active service is available.
+        def ensure_service!
+          raise "Must have active connection" unless service
+        end
+
+        ##
+        # Ensures the Google::Apis::BigqueryV2::Model object has been loaded
+        # from the service.
+        def ensure_gapi!
+          ensure_service!
+          return unless reference?
+          reload!
+        end
+
+        ##
+        # Fetch gapi and memoize whether resource exists.
+        def gapi_exists?
+          reload!
+          @exists = true
+        rescue Google::Cloud::NotFoundError
+          @exists = false
+        end
+
+        def patch_gapi! *attributes
+          return if attributes.empty?
+          ensure_service!
+          patch_args = Hash[attributes.map do |attr|
+            [attr, @gapi.send(attr)]
+          end]
+          patch_gapi = Google::Apis::BigqueryV2::Model.new patch_args
+          patch_gapi.model_reference = model_ref
+          @gapi = service.patch_model dataset_id, model_id, patch_gapi, etag
+
+          # TODO: restore original impl after acceptance test indicates that
+          # service etag bug is fixed
+          reload!
+        end
+
+        ##
+        # Load the complete representation of the model if it has been
+        # only partially loaded by a request to the API list method.
+        def ensure_full_data!
+          reload! unless resource_full?
+        end
+
+        ##
+        # Yielded to a block to accumulate changes for a patch request.
+        class Updater < Model
```

Comment:
If Updater isn't currently used, can we remove it (for now)?

Reply:
Removed.